Core/src/org/sleuthkit/autopsy/casemodule/Case.java

```diff
@@ -597,13 +598,32 @@ public static Case getCurrentCase() {
          * IllegalStateException; change to throw checked exception or return
          * null
          */
-        if (null != currentCase) {
-            return currentCase;
-        } else {
-            throw new IllegalStateException(NbBundle.getMessage(Case.class, "Case.getCurCase.exception.noneOpen"));
+        try {
+            Case curCase = getOpenCase(); 
+            return curCase;
+        } catch (NoCurrentCaseException e) {
+            throw new IllegalStateException(NbBundle.getMessage(Case.class, "Case.getCurCase.exception.noneOpen"), e);
         }
+        
     }
 
+    /**
+     * Gets the current open case, if there is one, at the time of the call.
+     *
+     * @return The open case.
+     *
+     * @throws NoCurrentCaseException if there is no open case.
+     */
+    public static Case getOpenCase() throws NoCurrentCaseException {
+        try {
+            Case openCase = new Case(currentCase.getCaseType(), 
+                                currentCase.getCacheDirectory(), 
+                                new CaseDetails(currentCase.getDisplayName(), currentCase.getNumber(), currentCase.getExaminer(), currentCase.getExaminerPhone(), currentCase.getExaminerEmail(), currentCase.getCaseNotes()));
+            return openCase;
```

Comment:
For this fix, we should be making a local copy of the reference in currentCase, checking the local reference for null, and finally returning the local reference. This does not prevent another thread from closing the case, but it does make things a bit better because we will at least not risk returning null and violating the contract of the method. Dealing with the possibility of the current case being closed while other objects have a reference to it is a much bigger problem for another story.